crates/bevy_asset/src/asset_server.rs

```diff
@@ -499,9 +507,11 @@ impl AssetServer {
                             .get_or_insert_with(|| self.server.asset_sources.write());
                         if let Some(source_info) = asset_sources.get_mut(&id.source_path_id()) {
                             source_info.committed_assets.remove(&id.label_id());
-                            if source_info.is_loaded() {
-                                source_info.load_state = LoadState::Loaded;
-                            }
+                            source_info.load_state = if source_info.is_loaded() {
+                                LoadState::Loaded
+                            } else {
+                                LoadState::NotLoaded
+                            };
```

Comment:
I'm curious if this call is necessary?
Internally, `is_loaded` check if `source_info.committed_assets.len() == self.meta.assets.len()`.
And if we're removing an asset via `source_info.committed_assets.remove(&id.label_id());`, this would always return false no?

If so, we can just set `source_info.load_state = LoadSate::NotLoaded`